scripts/auto-rebase/rebase.py

```diff
@@ -308,13 +327,16 @@ def main():
     pr_title = create_pr_title(rebase_branch_name, rebase_result.success)
     desc = generate_pr_description(rebase_branch_name, get_release_tag(release_amd), get_release_tag(release_arm), prow_job_url, rebase_result.success)
 
+    comment = ""
     pr = try_get_pr(gh_repo, org, base_branch, rebase_branch_name)
     if pr == None:
         pr = create_pr(gh_repo, base_branch, rebase_branch_name, pr_title, desc)
-        post_comment(pr)
     else:
         update_pr(pr, pr_title, desc)
-        post_comment(pr, f"Rebase job updated the branch\n{desc}")
+        comment = f"Rebase job updated the branch\n{desc}"
+
+    cleanup_branches(gh_repo)
```

Comment:
Now that we run this script for each branch, is there any concern about a race condition if we do this in every job? Should it be standalone?

Reply:
Or, maybe, delete only the stale branches in the current release where it is run.

Reply:
Or this logic could execute only when job runs for `main` branch (i.e. current release under development) but scan delete branches for all releases. This way we won't have to backport this immediately to 4.12 (which still uses old `create_pr.py` - at least until we think that `rebase.py` is good enough to backport it and don't touch anymour)

Reply:
Yes, I'm for limiting this functionality to the main branch - good idea

Reply:
I like the idea of having the special logic only in the main branch job.

Reply:
Changed